src/UIKit/UIAdaptivePresentationControllerDelegate.cs

```diff
@@ -0,0 +1,37 @@
+#if !WATCH
+using System;
+using XamCore.UIKit;
+using XamCore.ObjCRuntime;
+using XamCore.Foundation;
+
+namespace XamCore.UIKit {
+	public partial class UIAdaptivePresentationControllerDelegate
+	{
+		// this is a workaround to allow exposing the old API ()
+		[Export ("adaptivePresentationStyleForPresentationController:")]
+		public virtual UIModalPresentationStyle GetAdaptivePresentationStyle (UIPresentationController forPresentationController)
+		{
+			throw new You_Should_Not_Call_base_In_This_Method ();
+		}
+	}
+	
+	public static partial class UIAdaptivePresentationControllerDelegate_Extensions
+	{
+		public static UIModalPresentationStyle GetAdaptivePresentationStyle (this IUIAdaptivePresentationControllerDelegate This, UIPresentationController forPresentationController)
+		{
+			UIKit.UIApplication.EnsureUIThread ();
```

Comment:
^ the linker won't be able to remove the above check. I can only do so for generated code.